contracts/facade/FacadeRead.sol

```diff
@@ -28,26 +28,46 @@ contract FacadeRead is IFacadeRead {
     /// @return {qRTok} How many RToken `account` can issue given current holdings
     /// @custom:static-call
     function maxIssuable(IRToken rToken, address account) external returns (uint256) {
+        (address[] memory erc20s, ) = rToken.main().basketHandler().quote(FIX_ONE, FLOOR);
+        uint256[] memory balances = new uint256[](erc20s.length);
+        for (uint256 i = 0; i < erc20s.length; ++i) {
+            balances[i] = IERC20(erc20s[i]).balanceOf(account);
+        }
+        return maxIssuableByAmounts(rToken, balances);
+    }
+
+    /// @param amounts {qTok} The balances of each basket ERC20 to assume
+    /// @return {qRTok} How many RToken `account` can issue given current holdings
+    /// @custom:static-call
+    function maxIssuableByAmounts(IRToken rToken, uint256[] memory amounts)
+        public
+        returns (uint256)
+    {
         IMain main = rToken.main();
 
         require(!main.frozen(), "frozen");
 
         // Poke Main
         main.assetRegistry().refresh();
 
-        // {BU}
-        BasketRange memory basketsHeld = main.basketHandler().basketsHeldBy(account);
-        uint192 needed = rToken.basketsNeeded();
-
-        int8 decimals = int8(rToken.decimals());
-
-        // return {qRTok} = {BU} * {(1 RToken) qRTok/BU)}
-        if (needed.eq(FIX_ZERO)) return basketsHeld.bottom.shiftl_toUint(decimals);
+        // Get basket ERC20s
+        IBasketHandler bh = main.basketHandler();
+        if (!bh.isReady()) return 0;
```

Comment:
Do we need this if we're just quoting here?

Reply:
No we don't...